frontend/javascripts/oxalis/view/left-border-tabs/histogram_view.js

```diff
@@ -185,6 +193,80 @@ class Histogram extends React.PureComponent<HistogramProps, HistogramState> {
     500,
   );
 
+  getCuboidForViewport = async (viewport: OrthoView, layerName: string) => {
+    const state = Store.getState();
+
+    const [curX, curY, curZ] = Dimensions.transDim(
+      Dimensions.roundCoordinate(getPosition(state.flycam)),
+      viewport,
+    );
+    const [halfViewportExtentX, halfViewportExtentY] = getHalfViewportExtentsFromState(
+      state,
+      viewport,
+    );
+
+    const min = Dimensions.transDim(
+      V3.sub([curX, curY, curZ], [halfViewportExtentX, halfViewportExtentY, 0]),
+      viewport,
+    );
+    const max = Dimensions.transDim(
+      V3.add([curX, curY, curZ], [halfViewportExtentX, halfViewportExtentY, 1]),
+      viewport,
+    );
+
+    const resolutionIndex = getRequestLogZoomStep(state);
+    const cuboid = await api.data.getDataFor2DBoundingBox(layerName, { min, max }, resolutionIndex);
+    return cuboid;
+  };
+
+  getClippingValues = async (layerName: string, threshold: number = 0.05) => {
+    const cuboidXY = await this.getCuboidForViewport(OrthoViews.PLANE_XY, layerName);
+    const cuboidXZ = await this.getCuboidForViewport(OrthoViews.PLANE_XZ, layerName);
+    const cuboidYZ = await this.getCuboidForViewport(OrthoViews.PLANE_YZ, layerName);
+
+    const { elementClass } = getLayerByName(Store.getState().dataset, layerName);
+    const [TypedArrayClass] = getConstructorForElementClass(elementClass);
+    const cuboidData = new TypedArrayClass(cuboidXY.length + cuboidXZ.length + cuboidYZ.length);
+
+    cuboidData.set(cuboidXY);
+    cuboidData.set(cuboidXZ, cuboidXY.length);
+    cuboidData.set(cuboidYZ, cuboidXY.length + cuboidXZ.length);
+
+    const valueCounts = {};
+    let maxCount = 0;
+    for (let i = 0; i < cuboidData.length; i++) {
+      if (cuboidData[i] !== 0 && cuboidData[i] in valueCounts) {
+        valueCounts[cuboidData[i]] += 1;
+        if (maxCount < valueCounts[cuboidData[i]]) maxCount = valueCounts[cuboidData[i]];
+      } else valueCounts[cuboidData[i]] = 1;
+    }
+
+    let lowClip = 255;
```

Comment:
There are multiple data types which a layer can have. 255 is only valid for uint8. For uint16 you would need 2**16 - 1. However, I think, it's easier to initialize both values (`lowClip` and `highClip`) with `-1` and check explicitly against `-1` (i.e., `cuboidData[i] !== 0 && (cuboidData[i] < lowClip || lowClip === -1)`.

If the entire cube data is `0` then, low and max would still be -1. In that case, you wouldn't have nice default values (which is the advantage of your current approach), but I would argue that the histogram should not be changed at all in that case, since no useful data exists at this moment. You could simply `return null` in that case à  la:

```
// the following would replace the old "if (lowClip > highClip) return [highClip, lowClip];" line probably
if (lowClip === -1 || highClip === -1) {
  return null;
}
```

The caller would need to check whether the returned value is null and then simply do nothing (or even better: show a toast message that the clipping didn't succeed).

Reply:
Could there also be a problem with uint24 not being part of `TypedArrayClass`?
```
type t =
  | typeof Uint8Array
  | typeof Uint16Array
  | typeof Uint32Array
  | typeof Float32Array
```

Reply:
![Screenshot 2022-01-18 at 21 31 24](https://user-images.githubusercontent.com/46814136/150014031-60325de9-d837-4627-b581-4c68d6ed9853.png)

Reply:
> Could there also be a problem with uint24 not being part of `TypedArrayClass`?
> 
> ```
> type t =
>   | typeof Uint8Array
>   | typeof Uint16Array
>   | typeof Uint32Array
>   | typeof Float32Array
> ```

This should still work, since `uint24` is basically only an alias for "uint8 with 3 channels". The `getConstructorForElementClass` method which you are using will return `uint8` and `3` when the element class is "uint24". The channel count could be used to calculate the total length of the flattened array, but since you are using the `length` property of the other arrays, you don't respect the channel count, as far as I see.



> ![Screenshot 2022-01-18 at 21 31 24](https://user-images.githubusercontent.com/46814136/150014031-60325de9-d837-4627-b581-4c68d6ed9853.png)

Cool toast!